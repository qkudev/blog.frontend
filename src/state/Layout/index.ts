import * as Actions from './actions'

export * from './reducer'
export { Actions }
