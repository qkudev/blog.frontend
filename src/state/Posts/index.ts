import * as Actions from './actions'

export { Actions }
export * from './reducer'
