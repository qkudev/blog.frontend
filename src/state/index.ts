import { applyMiddleware, combineReducers, createStore, Middleware } from 'redux'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension'
import logger from 'redux-logger'

const isEnvProduction = process.env.NODE_ENV === 'production'

let middlewares: Middleware[] = [thunk]

if (!isEnvProduction) {
  middlewares.push(logger)
}

export interface IState {
}

export const initState: IState = {}

const rootReducer = combineReducers({})

export const store = createStore(
  rootReducer,
  initState,
  isEnvProduction
    ? applyMiddleware(...middlewares)
    : composeWithDevTools(applyMiddleware(...middlewares))
)
