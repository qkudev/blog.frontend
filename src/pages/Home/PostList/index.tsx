import * as React from 'react'
import InfiniteScroller from 'react-infinite-scroller'
import { connect } from 'react-redux'
import * as State from '../../../state'
import { Post, Spinner } from '../../../components'
import * as API from '../../../api'

type Props = State.Posts.IState

function FailMessage() {
  return <h1 style={{ textAlign: 'center' }}>Failed to fetch posts :(</h1>
}

class PostList extends React.Component<Props> {
  render() {
    const { loading, error, allIds, byId, pagination } = this.props
    if (error) {
      return <FailMessage />
    }
    return (
      <InfiniteScroller
        loadMore={API.getPosts}
        hasMore={
          !loading &&
          !error &&
          (!pagination || pagination.page < pagination.pages)
        }
        loader={<Spinner key='spinner' />}
      >
        {allIds.map(id => (
          <Post key={id} {...byId[id]} />
        ))}
      </InfiniteScroller>
    )
  }
}

const mapStateToProps = (state: State.IState) => ({
  ...state.posts
})

export default connect(mapStateToProps)(PostList)
