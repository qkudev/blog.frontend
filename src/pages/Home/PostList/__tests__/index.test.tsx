import * as React from 'react'
import renderer from 'react-test-renderer'
import { store, Provider, Posts } from '../../../../state'
import PostList from '../index'

const testPosts: IPost[] = [
  {
    _id: 'qwerty',
    body: 'Some text',
    updatedAt: '2019-04-14T23:41:36.340Z',
    createdAt: '2019-04-14T23:41:36.340Z',
    __v: 0
  }
]

const testPagination: Pagination = {
  page: 1,
  limit: 10,
  total: 0,
  pages: 1
}

const testError = 'TestError'

describe('Components::PostList', function() {
  it('should match snapshot with loading state', function() {
    store.dispatch(Posts.Actions.getPostsBegin())

    const postList = renderer.create(
      <Provider store={store}>
        <PostList {...store.getState().posts} />
      </Provider>
    )
    let tree = postList.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with success state', function() {
    store.dispatch(Posts.Actions.getPostsSuccess(testPosts, testPagination))

    const postList = renderer.create(
      <Provider store={store}>
        <PostList {...store.getState().posts} />
      </Provider>
    )
    let tree = postList.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with fail state', function() {
    store.dispatch(Posts.Actions.getPostsFail(testError))

    const postList = renderer.create(
      <Provider store={store}>
        <PostList {...store.getState().posts} />
      </Provider>
    )
    let tree = postList.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
