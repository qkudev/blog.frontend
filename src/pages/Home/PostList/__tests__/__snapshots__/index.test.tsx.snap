// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components::PostList should match snapshot with fail state 1`] = `
<h1
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  Failed to fetch posts :(
</h1>
`;

exports[`Components::PostList should match snapshot with loading state 1`] = `<div />`;

exports[`Components::PostList should match snapshot with success state 1`] = `
<div>
  <div
    className="post"
  >
    <div
      className="post-body"
    >
      <p>
        Some text
      </p>
    </div>
  </div>
</div>
`;
